From 6770107f24a14e944cd1c9c097db416f58431036 Mon Sep 17 00:00:00 2001
From: Jari Aalto <jari.aalto@cante.net>
Date: Thu, 17 Dec 2009 14:20:10 +0200
Subject: [PATCH] Makefile: Change name. Fix linking with GTK+

Signed-off-by: Jari Aalto <jari.aalto@cante.net>
---
 Makefile |   32 ++++++++++++++++++++------------
 1 files changed, 20 insertions(+), 12 deletions(-)

diff --git a/Makefile b/Makefile
index 219e871..678fda4 100644
--- a/Makefile
+++ b/Makefile
@@ -1,4 +1,15 @@
-all:
+
+LDFLAGS  = -Wl,--no-add-needed -Wl,--no-undefined
+INCFLAGS = -I/usr/include/gtk-2.0
+LIBS	 = -lX11
+
+# The listing "pkg-config --libs gtk+-2.0" can be trimmed down to needed libs
+GTKLIBS  =  $(shell perl -e' print join qq( ), grep /(-x11|gobject|glib)/, @ARGV' -- $$(pkg-config --libs gtk+-2.0) )
+GTKFLAGS = `pkg-config --cflags gtk+-2.0`
+
+all: dock shutdown
+
+help:
 	@echo "wmShutdown version 0.1 - Rafael V. Aroca <rafael@linuxqos.cjb.net>"
 	@echo "Order of makes to install: dock, shutdown, install"
 
@@ -8,22 +19,19 @@ clean:
 	rm -f Shutdown
 
 install:
-	install wmShutdown /usr/local/bin
-	install wmShutdown.xpm /usr/share/pixmaps/
-	install -g root -o root Shutdown /usr/local/bin
-	chmod +s /sbin/shutdown
-	chmod +s /usr/local/bin/Shutdown
-	@echo Installed. Just type wmShutdown to use it
+	install -m 755 wmShutdown $(DESTDIR)/usr/bin/wmshutdown
+	install -m 644 wmShutdown.xpm $(DESTDIR)/usr/share/pixmaps/wmshutdown.xpm
+	install -m 755 Shutdown $(DESTDIR)/usr/bin/wmshutdown-run
 
 uninstall:
-	rm -f /usr/local/bin/wmShutdown
-	rm -f /usr/local/bin/Shutdown
+	rm -f /bin/wmShutdown
+	rm -f /bin/Shutdown
 	rm -f /usr/share/pixmaps/wmShutdown.xpm
 	@echo ":-( Uninstalled."
 
 dock: wmShutdown.c
-	$(CC) -c -o wmShutdown.o wmShutdown.c `gtk-config --cflags` 
-	$(CC) -o wmShutdown wmShutdown.o `gtk-config --libs` 
+	$(CC) $(INCFLAGS) $(GTKFLAGS) -c -o wmShutdown.o wmShutdown.c
+	$(CC) $(LDFLAGS) -o wmShutdown wmShutdown.o $(GTKLIBS) $(LIBS)
 
 shutdown: shutdown.c
-	$(CC) -o Shutdown shutdown.c 
+	$(CC) $(LDFLAGS) $(INCFLAGS) -o Shutdown shutdown.c
-- 
1.6.5

