From 0eda3e6215b8b2fa09fe032861313d08a98bb490 Mon Sep 17 00:00:00 2001
From: Jari Aalto <jari.aalto@cante.net>
Date: Tue, 8 Dec 2009 14:27:05 +0200
Subject: [PATCH] wmShutdown.c: change names

Signed-off-by: Jari Aalto <jari.aalto@cante.net>
---
 wmShutdown.c |    6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/wmShutdown.c b/wmShutdown.c
index 7f7e14e..f03f4b9 100644
--- a/wmShutdown.c
+++ b/wmShutdown.c
@@ -57,7 +57,7 @@ int desliga(void) {
 	FILE *output;
 	char *teste;
 
-	output = popen("/usr/local/bin/Shutdown -h", "r");
+	output = popen("/usr/bin/wmshutdown -h", "r");
 	teste = (char*) fgetc(output);
 	while ((int)teste != EOF) {
 		g_print("%c", teste);
@@ -70,7 +70,7 @@ int reinicia(void) {
 	FILE *output;
 	char *teste;
 
-	output = popen("/usr/local/bin/Shutdown -r", "r");
+	output = popen("/usr/bin/wmshutdown-run -r", "r");
 	teste = (char*) fgetc(output);
 	while ((int)teste != EOF) {
 		g_print("%c", teste);
@@ -148,7 +148,7 @@ int main(int argc, char *argv[]) {
 	gtk_widget_realize(dockArea);	    
 
         icon = (gpointer) gdk_pixmap_create_from_xpm (gtkiw->window, &mask,
-                	NULL, "/usr/share/pixmaps/wmShutdown.xpm");
+                	NULL, "/usr/share/pixmaps/wmshutdown.xpm");
 
         pixmap = gtk_pixmap_new((gpointer) icon, mask);
         gtk_widget_show(pixmap);
-- 
1.6.5

